aten/src/ATen/Context.h

```diff
@@ -369,10 +369,32 @@ class TORCH_API Context {
   bool _deterministic_algorithms = false;
   bool _deterministic_algorithms_warn_only = false;
   bool _deterministic_fill_uninitialized_memory = true;
+
+#ifdef USE_CUDA
+#ifdef USE_FLASH_ATTENTION
   bool enabled_flashSDP = true;
+#else
+  bool enabled_flashSDP = false;
+#endif
+
+#ifdef USE_MEM_EFF_ATTENTION
   bool enabled_mem_efficientSDP = true;
-  bool enabled_mathSDP = true;
+#else
+  bool enabled_mem_efficientSDP = false;
+#endif
+
+#ifdef USE_CUDNN_ATTENTION
   bool enabled_cudnnSDP = true;
+#else
+  bool enabled_cudnnSDP = false;
+#endif
+#else
+  bool enabled_flashSDP = true;
+  bool enabled_mem_efficientSDP = true;
+  bool enabled_cudnnSDP = true;
```

Comment:
is this supposed to be true for !USE_CUDA?

Reply:
yeah, the semantics of this op flag are supposed to be "did the user specifically disable this backend" when we didnt build with support for this backend we shouldnt enable it by default. 
They could still set it to true if they want to. All this flag does is add another filter to sdp_utils "did the user say I am allowed to use this backend?". 

On CPU, we go through this func: https://github.com/pytorch/pytorch/blob/efefea52e0b3c8e356c391dc531987531b93a172/aten/src/ATen/native/transformers/sdp_utils_cpp.cpp#L60

so really only flash_attention flag would do anything and we want to keep that default value to True